### PR TITLE
Document removing data nodes

### PR DESCRIPTION
**Pull request by rwaight**
Document guidance on the removal / decommissioning of data nodes in `adding-removing-nodes.asciidoc`

Closes https://github.com/elastic/elasticsearch/issues/46338

### CODE REVIEW

**Comment by lcawl** [comment]
You can preview the changes here:
http://elasticsearch_46355.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.3/modules-discovery-adding-removing-nodes.html

As a drive-by comment, I wonder if this information shouldn't be migrated here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/add-elasticsearch-nodes.html

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 1]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
 DELETE /_cluster/voting_config_exclusions?wait_for_removal=false
 --------------------------------------------------
 // CONSOLE
+
+[[modules-discovery-removing-data-nodes]]
+==== Removing data nodes [WIP // DRAFT]
+When removing data nodes, it is important not to remove too many at the same time. The way of decommissioning a node is to instruct Elasticsearch to move shards away from that node. You just need to make sure that you have enough room in the cluster to accommodate the shards. Also remember to wait for  Elasticsearch to move shards around before shutting down the node.
+
+1. Validate node settings, including allocation <<allocation-awareness>> configurations
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: It isn't clear here just what you need to "validate". I'd provide a bit more detail--something like, "If you use custom node attributes to control where Elasticsearch allocates shards, make sure that those awareness attributes are configured on other nodes in the cluster."

.

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 2]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
...
+2. Ensure the <<cluster-health>> is green
+3. Tell the cluster to move shards away (exclude) from the node being removed
++
+[source,js]
+--------------------------------------------------
+# Do not allocate shards by moving shards away (excluding) from the node
+# Define `cluster.routing.allocation.exclude.{attribute}` using one of the 
+# supported attributes:  `_name`, `_ip`, and `_host`
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: In general, we avoid including comments in the example snippets in favor of incorporating the content into the intro or using callouts. It's not necessary to repeat what can easily be gleaned by reading the example itself.

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 3]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
...
+PUT _cluster/settings
+{
+  "transient" : {
+    "cluster.routing.allocation.exclude._name" : "name_of_node_being_removed"
+  }
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: Update the node name to match the lead-in:

`    "cluster.routing.allocation.exclude._name" : "prod-data-2"`

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 4]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
...
+}
+--------------------------------------------------
+// CONSOLE
++
+4. Monitor the cluster to ensure all primary shards have been allocated onto other nodes
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: The past-tense doesn't quite fit here. I'd say, "Monitor the cluster to ensure all primary shards are successfully migrated to other nodes."

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 5]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
...
+5. Once the cluster is healthy and all shards have been reallocated to other nodes, turn off the elasticsearch service of the node being removed, then the node can be shut down/decommissioned
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: It's not necessary to reiterate the previous step. I'd just say, "Once the migration is complete and the cluster state is green, you can safely shut down the node."

File: docs/reference/modules/discovery/adding-removing-nodes.asciidoc [hunk 6]
```diff
@@ -135,3 +135,28 @@ DELETE /_cluster/voting_config_exclusions
...
 // CONSOLE
+
+[[modules-discovery-removing-data-nodes]]
+==== Removing data nodes [WIP // DRAFT]
+When removing data nodes, it is important not to remove too many at the same time. The way of decommissioning a node is to instruct Elasticsearch to move shards away from that node. You just need to make sure that you have enough room in the cluster to accommodate the shards. Also remember to wait for  Elasticsearch to move shards around before shutting down the node.
+
+1. Validate node settings, including allocation <<allocation-awareness>> configurations
+2. Ensure the <<cluster-health>> is green
+3. Tell the cluster to move shards away (exclude) from the node being removed
++
```
Comment (debadair) [inline docs/reference/modules/discovery/adding-removing-nodes.asciidoc]: I'd probably say something like, "Update the cluster's shard allocation settings to exclude the node you want to remove. This instructs Elasticsearch to migrate shards off of the node. You can specify the node by name, IP address, or host. For example, the following request excludes `prod-data-2`:

**Review by debadair — CHANGES_REQUESTED** [review]
Left some suggestions. I'd go ahead and add this information here for now, but avoid using the modules prefix in the anchor.

**Comment by colings86** [comment]
@rwaight  are you still wanting to progress this PR or should it be closed?

**Comment by colings86** [comment]
I'm going to close this PR since there has been no activity in 11 months.

**Pull request closed without merging**